src/Microsoft.TestPlatform.CoreUtilities/Helpers/EnvironmentVariableHelper.cs

```diff
@@ -13,8 +13,37 @@ namespace Microsoft.VisualStudio.TestPlatform.CoreUtilities.Helpers;
 
 internal class EnvironmentVariableHelper : IEnvironmentVariableHelper
 {
+    /// <inheritdoc />
     public string GetEnvironmentVariable(string variable)
         => Environment.GetEnvironmentVariable(variable);
+
+    /// <inheritdoc />
+    // TODO: This helper won't be needed when we will stop support for .NET Standard 1.3 and UWP
+    public TEnum GetEnvironmentVariableAsEnum<TEnum>(string variable, TEnum defaultValue = default) where TEnum : Enum
+    => Environment.GetEnvironmentVariable(variable) is string value && !string.IsNullOrEmpty(value)
+        ? (TEnum)Enum.Parse(typeof(TEnum), value)
+        : defaultValue;
+
+    /// <inheritdoc />
+    public T GetEnvironmentVariable<T>(string variable, T defaultValue = default) where T : IConvertible
+        => Environment.GetEnvironmentVariable(variable) is string value && !string.IsNullOrEmpty(value)
+            ? ConvertToType<T>(value)
+            : defaultValue;
+
+    private static T ConvertToType<T>(string input)
+    {
+        var targetType = typeof(T);
+        var conversionType = Nullable.GetUnderlyingType(targetType) ?? targetType;
+
+#if !WINDOWS_UWP && !NETSTANDARD1_3
+        if (conversionType.IsEnum)
+        {
+            return (T)Enum.Parse(conversionType, input);
+        }
+#endif
+
+        return (T)Convert.ChangeType(input, conversionType);
```

Comment:
Won't this throw quite easily?

Reply:
It's a lot less safe than doing `XXX.TryParse` for sure. Also we are not allowing any format provider. I can add a try/catch but that would still be less good than specific parse call.

Maybe this helper is not a good idea and I should keep only the enum one (replacing `Parse` with `TryParse`).